modules/internal/ChapelDistribution.chpl

```diff
@@ -228,8 +240,11 @@ module ChapelDistribution {
         local {
           _arrsLock.lock();
           _arrs_containing_dom -=1;
-          if rmFromList && trackArrays() then
-            _arrs.remove(x);
+          if rmFromList && trackArrays() {
+            if _arrs_head == x then _arrs_head = x.next;
+            if x.next != nil then x.next!.prev = x.prev;
+            if x.prev != nil then x.prev!.next = x.next;
+          }
```

Comment:
I'd consider setting `x.next` and `x.prev` to `nil` here

Reply:
If we have the assert for add, do you think that's needed? This is only used shortly before deleting an array.

Reply:
assert for add is sufficient thanks